CollapseLauncher/XAMLs/MainApp/Pages/SettingsPage.xaml.cs

```diff
@@ -1397,6 +1408,94 @@ private static double DownloadChunkSize
+
+        private async void ValidateAndApplyDnsSettings(object sender, RoutedEventArgs e)
+        {
+            if (sender is not Button senderAsButton)
+            {
+                return;
+            }
+
+            senderAsButton.IsEnabled = false;
+            try
+            {
+                DnsSettingsTestTextChecking.Visibility = Visibility.Visible;
+
+                string?           dnsHost     = _dnsSettingsContext.ExternalDnsAddresses;
+                DnsConnectionType connType    = (DnsConnectionType)_dnsSettingsContext.ExternalDnsConnectionType;
+                string            dnsSettings = $"{dnsHost}|{connType}";
+
+                (bool isSuccess, string[]? resultHosts) resultHosts = await Task.Factory.StartNew(() =>
+                {
+                    bool isSuccess =
+                        HttpClientBuilder.TryParseDnsHosts(dnsSettings, true, true,
+                                                           out string[]? resultHosts);
+                    return (isSuccess, resultHosts);
+                }, TaskCreationOptions.DenyChildAttach);
+
+                if (!resultHosts.isSuccess)
+                {
+                    throw new InvalidOperationException($"The current DNS host string: {dnsSettings} has malformed separator or one of the hostname's IPv4/IPv6 cannot be resolved! " + 
+                                                        $"Also, make sure that you use one of these separators: {_dnsSettingsSeparatorList}");
+                }
+
+
+                (bool isSuccess, DnsConnectionType resultConnType) resultConnType = await Task.Factory.StartNew(() =>
+                {
+                    bool isSuccess =
+                        HttpClientBuilder.TryParseDnsConnectionType(dnsSettings, out DnsConnectionType resultConnType);
+                    return (isSuccess, resultConnType);
+                }, TaskCreationOptions.DenyChildAttach);
+
+                if (!resultConnType.isSuccess)
+                {
+                    DnsConnectionType[] types       = Enum.GetValues<DnsConnectionType>();
+                    string              typesInList = string.Join(", ", types);
+                    throw new InvalidOperationException($"The current DNS host string: {dnsSettings} has no valid DNS Connection Type. " + 
+                                                        $"The valid values are: {typesInList}");
+                }
+
+                const string testUrl = "https://gitlab.com/bagusnl/CollapseLauncher-ReleaseRepo/-/raw/main/LICENSE";
+
+                HttpClient httpClientWithCustomDns = new HttpClientBuilder<SocketsHttpHandler>()
+                                                    .UseLauncherConfig(skipDnsInit: true)
+                                                    .UseExternalDns(resultHosts.resultHosts, resultConnType.resultConnType)
```

Comment:
As I have explained in the Discord, our default HttpClient timeout (90s) might be too long for a simple check sequence, so I propose to have override just for this test sequence to add `.SetTimeout(TimeSpan.FromSeconds(10))` or even 5 to ensure that user get feedback of their settings fast

Reply:
Done. Setting the timeout to 5 secs